src/main/kotlin/skytils/skytilsmod/features/impl/dungeons/ScoreCalculation.kt

```diff
@@ -179,12 +179,13 @@ object ScoreCalculation {
     }
     val speedScore = totalElapsed.map { time ->
         when {
-            time <= 480.0 -> 100.0
-            time <= 580.0 -> 148 - 0.1 * time
-            time <= 980 -> 119 - 0.05 * time
-            time < 3060 -> 102 - 1.0 / 30.0 * time
+            time < 492.0 -> 100.0
+            time < 600.0 -> 140 - time / 12
+            time < 840.0 -> 115 - time / 24
+            time < 1140.0 -> 108 - time / 30
+            time < 3940.0 -> 98.5 - 1.0 / 40.0 * time
```

Comment:
This should just be `98.5 - time / 40`

Reply:
Lines 183~185 are meant to be integer divisions while line 186 does care about floating-point precisions:
if time is actually a floating number, then change scores in lines 183~185 from "time / [number]" to "time.toInt() / [number]" or whatever that forces integer division(along with your suggestion).
The reason behind integer divisions is Skyblock Dungeons stripping the remainder of speed score timestamps rather than rounding up/down accordingly(For example, a 10m 13s run, a 10m 18s run, and a 10m 22s run in F3 all give 99 speed pts even if speed score timestamps are actually 100/99/98 for 10m 0s/10m 12s/10m 24s[lmk if that's no longer the case anymore]).
Feel free to make changes as you see fit ig(I'm new to Kotlin after all).